app/views/devise/registrations/edit.html.erb

```diff
@@ -1,4 +1,4 @@
-<h2>Edit <%= resource_name.to_s.humanize %></h2>
+<h2>Modifier mes informations </h2>
```

Comment:
You can use the devise-i18n gem to translate those views (re generate the views): https://github.com/tigrish/devise-i18n